### PR TITLE
Feature/1471 bsc chain

### LINKED ISSUE #1471: Add BSC network
### Adding BSC to Swapr includes:

SDK:

- add basic config (chainId, native currency, core tokens, etc.)
- investigate which DEXes can be ecoRouted and include them. Probably use pancakeSwap.

FE:

- enable adding/switching network to BSC
- disable Swapr specific features when connected to BSC (disable Liquidity, Rewards)

Bridge:

- enable BSC support on Socket
- add bridgable tokens for BSC

### PR DESCRIPTION
**Pull request by jorgedclopes**
# Summary

**Partially** Fixes #1471

- Add BSC chain suppport.
- Add PancakeSwap as a usable exchange.
- Add SushiSwap to BSC chain.

Bridge is not functional yet. Tokens are not yet mapped.

# To Test
Ignore the automated deployment.

1. Locally, clean your environment (node_modules & yarn.lock). Link the sdk from [this branch](https://github.com/levelkdev/swapr-sdk/tree/feature/1471-BSC-network)
2. Start the application and test the functionalities
3. Test the several functionalities. You should be able to switch to the BSC chain and be able to swap.

Note: The SDK dependency will be reverted to a standard format once we have an SDK release.

### CODE REVIEW

**Comment by 0xVenky** [comment]
Hey, dont give me high hopes that I have made it in my life.

image

**Comment by niemam29** [comment]
I have noticed that after adding bsc network selectors in bridge are too long for full hd devices 
cach2

File: src/constants/lists.ts [hunk 1]
```diff
@@ -22,6 +22,10 @@ const BAOSWAP_LIST = 'https://raw.githubusercontent.com/baofinance/tokenlists/ma
 const QUICKSWAP_LIST = 'https://unpkg.com/quickswap-default-token-list@1.2.29/build/quickswap-default.tokenlist.json'
 const DFYN_LIST = 'https://raw.githubusercontent.com/dfyn/new-host/main/list-token.tokenlist.json'
 const OPTIMISM_LIST = 'https://static.optimism.io/optimism.tokenlist.json'
+const PANCAKESWAP_LIST =
+  'https://raw.githubusercontent.com/pancakeswap/pancake-toolkit/master/packages/token-lists/lists/pancakeswap-default.json'
+const SUSHISWAP_LIST =
```
Comment (Diogomartf) [inline src/constants/lists.ts:27]: why we also added sushi swap?

File: src/lib/eco-router/api.ts [hunk 2]
```diff
@@ -195,6 +195,7 @@ export async function getExactIn(
   ])
   const unsortedTrades = unsortedTradesWithUndefined.filter((trade): trade is Trade => !!trade)
 
+  console.log({ uniswapV2TradesList, unsortedTradesWithUndefined, unsortedTrades })
```
Comment (Diogomartf) [inline src/lib/eco-router/api.ts:198]: do we want this log?

File: src/constants/lists.ts (hunk 1 above)
Comment (jorgedclopes) [reply src/constants/lists.ts:27]: I added sushiswap as a usable exchange in the BSC chain. As a consequence I thought I could also add its list.
Reviewing this, I am facing with a format problem. This list is not similar to other lists... as in, it doesn't include metadata information and the tokens aren't under a "tokens" key.
I might remove it for now.

File: src/apollo/client.ts [hunk 3]
```diff
@@ -13,6 +13,8 @@ export const subgraphClientsUris: { [chainId in SWPRSupportedChains]: string } =
   [ChainId.GOERLI]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-goerli',
   [ChainId.ARBITRUM_RINKEBY]: 'https://api.thegraph.com/subgraphs/name/dxgraphs/swapr-arbitrum-rinkeby-v2',
   [ChainId.ARBITRUM_GOERLI]: '', // FIXME: fix this once the subgraph is deployed
+  [ChainId.BSC_MAINNET]: 'https://api.thegraph.com/subgraphs/name/unlock-protocol/bsc',
+  [ChainId.BSC_TESTNET]: 'https://api.thegraph.com/subgraphs/name/totemfi/staking-bsc-testnet',
```
Comment (adamazad) [inline src/apollo/client.ts:17]: Swapr core contracts are not deployed on BSC. These records should be empty. Modify `SWPRSupportedChains` to exclude BSC and its testnet.

File: src/apollo/client.ts [hunk 4]
```diff
@@ -34,6 +36,8 @@ export const subgraphClients: {
   [ChainId.GOERLI]: setupApolloClient(ChainId.GOERLI),
   [ChainId.ARBITRUM_RINKEBY]: setupApolloClient(ChainId.ARBITRUM_RINKEBY),
   [ChainId.ARBITRUM_GOERLI]: setupApolloClient(ChainId.ARBITRUM_GOERLI), // FIXME: fix this once the subgraph is deployed
+  [ChainId.BSC_MAINNET]: setupApolloClient(ChainId.BSC_MAINNET),
+  [ChainId.BSC_TESTNET]: setupApolloClient(ChainId.BSC_TESTNET),
```
Comment (adamazad) [inline src/apollo/client.ts:40]: Same here :)

File: src/apollo/client.ts [hunk 5]
```diff
@@ -90,4 +98,6 @@ export const subgraphBlocksClients: {
   [ChainId.GOERLI]: setupBlocksApolloClient(ChainId.GOERLI),
   [ChainId.ARBITRUM_RINKEBY]: setupBlocksApolloClient(ChainId.ARBITRUM_RINKEBY),
   [ChainId.ARBITRUM_GOERLI]: setupBlocksApolloClient(ChainId.ARBITRUM_GOERLI), // FIXME: fix this once the subgraph is deployed
+  [ChainId.BSC_MAINNET]: setupBlocksApolloClient(ChainId.BSC_MAINNET),
+  [ChainId.BSC_TESTNET]: setupBlocksApolloClient(ChainId.BSC_TESTNET),
```
Comment (adamazad) [inline src/apollo/client.ts:102]: Same here.

File: src/apollo/client.ts [hunk 6]
```diff
@@ -44,6 +48,8 @@ export const immediateSubgraphClients: { [chainId in SWPRSupportedChains]: Graph
   [ChainId.ARBITRUM_ONE]: new GraphQLClient(subgraphClientsUris[ChainId.ARBITRUM_ONE]),
   [ChainId.ARBITRUM_RINKEBY]: new GraphQLClient(subgraphClientsUris[ChainId.ARBITRUM_RINKEBY]),
   [ChainId.ARBITRUM_GOERLI]: new GraphQLClient(subgraphClientsUris[ChainId.ARBITRUM_GOERLI]), // FIXME: fix this once the subgraph is deployed
+  [ChainId.BSC_MAINNET]: new GraphQLClient(subgraphClientsUris[ChainId.BSC_MAINNET]),
+  [ChainId.BSC_TESTNET]: new GraphQLClient(subgraphClientsUris[ChainId.BSC_TESTNET]),
```
Comment (adamazad) [inline src/apollo/client.ts:52]: Same here.

File: src/constants/index.tsx [hunk 7]
```diff
@@ -125,6 +141,13 @@ export const USDT: { [key: number]: Token } = {
     'USDT',
     'Tether USD'
   ),
+  [ChainId.BSC_MAINNET]: new Token(
+    ChainId.BSC_MAINNET,
+    '0x55d398326f99059ff775485246999027b3197955',
+    6,
+    'USDT',
+    'Binance-Peg BSC-USD (BSC-USD)'
+  ),
```
Comment (adamazad) [inline src/constants/index.tsx:150]: Weren't these defined in the SDK? we should probably re-import from the SDK.

File: src/constants/index.tsx [hunk 8]
```diff
@@ -539,6 +584,28 @@ export const NETWORK_DETAIL: { [chainId: number]: NetworkDetails } = {
...
+  [ChainId.BSC_TESTNET]: {
+    chainId: `0x${ChainId.BSC_TESTNET.toString(16)}`,
+    chainName: 'BSC Testnet',
+    nativeCurrency: {
+      name: Currency.BNB.name || 'Binance Coin',
+      symbol: Currency.BNB.symbol || 'BNB',
+      decimals: Currency.BNB.decimals || 18,
+    },
+    rpcUrls: ['https://goerli.optimism.io'],
+    blockExplorerUrls: ['https://data-seed-prebsc-1-s1.binance.org:8545/'],
```
Comment (adamazad) [inline src/constants/index.tsx:607]: Wrong RPC url :)

File: src/constants/index.tsx [hunk 9]
```diff
@@ -539,6 +584,28 @@ export const NETWORK_DETAIL: { [chainId: number]: NetworkDetails } = {
...
+  [ChainId.BSC_MAINNET]: {
+    chainId: `0x${ChainId.BSC_MAINNET.toString(16)}`,
+    chainName: 'BSC Mainnet',
+    nativeCurrency: {
+      name: Currency.BNB.name || 'Binance Coin',
+      symbol: Currency.BNB.symbol || 'BNB',
+      decimals: Currency.BNB.decimals || 18,
+    },
+    rpcUrls: ['https://bsc-dataseed.binance.org'],
+    blockExplorerUrls: ['https://testnet.bscscan.com'],
```
Comment (adamazad) [inline src/constants/index.tsx:596]:
```suggestion
    blockExplorerUrls: ['https://bscscan.com'],
```

**Review by adamazad — CHANGES_REQUESTED** [review]
Great job. Overall, the implementation looks good. The UI might have to adapt to longer chain lists as @niemam29 pointed out. 

Address the comments, and it's a good to go from my side.

**Comment by jorgedclopes** [comment]
Netlify is failing because the link for swapr/sdk is pointing to the repo and not using the npm native process.
Either add '--network-concurrency 1' to the installation step on Netlify or a release of the SDK is needed for these to pass.

**Pull request merged**